src/components/shared/map/MapCenter.tsx

```diff
@@ -0,0 +1,170 @@
+      console.log('Center saved: ' + JSON.stringify([center.lat, center.lng]));
+      setIsButtonDisabled(true); // Grey out the Save button
+      setMarkerPosition([center.lat, center.lng]); // Drop pin at the center
+    }
+  };
+
+  const handleMapReady = () => {
+    const map = mapRef.current;
+    if (map) {
+      map.on('movestart', () => {
+        setIsButtonVisible(false); // Hide the Save button on map move
+        setMarkerPosition(null); // Remove the dropped pin when the user starts moving the map again
+      });
+      map.on('moveend', () => {
+        const center = map.getCenter();
+        setMapCenter([center.lat, center.lng]);
+        setIsButtonDisabled(false); // Enable the Save button on move
+        setIsButtonVisible(true); // Show the Save button when map movement stops
+      });
+    }
+  };
+
+  const typeText = (text: string, speed: number) => {
+    let i = 0;
+    clearInterval(intervalRef.current);  // Clear any existing interval before starting a new one
+    intervalRef.current = setInterval(() => {
+      if (i < text.length) {
+        setTypedMessage(prev => prev + text.charAt(i));  // Append the next character
+        i++;
+      } else {
+        clearInterval(intervalRef.current);  // Clear the interval once the text is fully typed
+      }
+    }, speed);
+  };
+
+  const closePopup = () => {
+    setShowPopup(false);
+    setPopupDismissed(true);
+    setIsButtonVisible(true); // Show the Save button when the popup is dismissed
+  };
+
+  return (
+    <div style={{ position: 'relative', height: '100vh', width: '100%' }}>
+      <MapContainer
+        center={mapCenter}
+        zoom={13}
+        style={{ height: '100%', width: '100%' }}
+        whenReady={() => handleMapReady()}
+      >
+        <TileLayer
+          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
+          attribution='Map data © OpenStreetMap contributors'
+        />
+        {markerPosition && <CenterMarker position={markerPosition} />}
+      </MapContainer>
+      {isButtonVisible && (
+        <div className="map-controls__item">
+          <button
+            className="map-controls__button"
+            onClick={saveCenterToLocalStorage}
+            role="button"
+            tabIndex={0}
+            disabled={isButtonDisabled}
+          >
+            Set Map Center
+          </button>
+        </div>
+      )}
+      {showPopup && (
+        <section
+          className="popup-message"
+          role="alertdialog"
+          aria-modal="true"
+          aria-labelledby="popupMessageTitle"
+          tabIndex={0}
+        >
+          <figure className="popup-message__logo">
+            <figcaption className="visually-hidden">App Name</figcaption>
+          </figure>
+          <div id="popupMessageTitle">
+            <h2 className="popup-message__title">Dear user</h2>
+            <p className="popup-message__typed-message">{typedMessage}</p>
+          </div>
+          <button
```

Comment:
You can use the shared Buttons component here that is referenced from shared/Forms/Buttons/Buttons.tsx.